db/background_mgr_attachment_migration.go

```diff
@@ -216,46 +220,49 @@ func (a *AttachmentMigrationManager) Run(ctx context.Context, options map[string
 			}
 		}
 		db.RequireAttachmentMigration = collectionsRequiringMigration
-
-		base.InfofCtx(ctx, base.KeyAll, "[%s] Finished migrating attachment metadata from sync data to global sync data. %d/%d docs changed", migrationLoggingID, a.DocsChanged.Value(), a.DocsProcessed.Value())
+		msg := fmt.Sprintf("[%s] Finished migrating attachment metadata from sync data to global sync data. %d/%d docs changed", migrationLoggingID, a.docsChanged.Load(), a.docsProcessed.Load())
+		failedDocs := a.docsFailed.Load()
+		if failedDocs > 0 {
+			msg += fmt.Sprintf(" with %d docs failed", failedDocs)
+		}
+		base.InfofCtx(ctx, base.KeyAll, msg)
 	case <-terminator.Done():
 		err = dcpClient.Close()
 		if err != nil {
 			base.WarnfCtx(ctx, "[%s] Failed to close attachment migration DCP client after attachment migration process was terminated %v", migrationLoggingID, err)
 			return err
 		}
-		if processFailure != nil {
-			return processFailure
-		}
 		err = <-doneChan
 		if err != nil {
 			return err
 		}
-		base.InfofCtx(ctx, base.KeyAll, "[%s] Attachment Migration was terminated. Docs changed: %d Docs Processed: %d", migrationLoggingID, a.DocsChanged.Value(), a.DocsProcessed.Value())
+		msg := fmt.Sprintf("[%s] Attachment Migration was terminated. %d/%d docs changed", migrationLoggingID, a.docsChanged.Load(), a.docsProcessed.Load())
+		failedDocs := a.docsFailed.Load()
+		if failedDocs > 0 {
+			msg += fmt.Sprintf(" with %d docs failed", failedDocs)
+		}
+		base.InfofCtx(ctx, base.KeyAll, msg)
 	}
 	return nil
 }
 
-func (a *AttachmentMigrationManager) SetStatus(docChanged, docProcessed int64) {
-
-	a.DocsChanged.Set(docChanged)
-	a.DocsProcessed.Set(docProcessed)
-}
-
-func (a *AttachmentMigrationManager) SetCollectionIDs(collectionID []uint32) {
+// setCollectionIDs sets the collection IDs that are undergoing migration.
+func (a *AttachmentMigrationManager) setCollectionIDs(collectionIDs []uint32) {
 	a.lock.Lock()
 	defer a.lock.Unlock()
-
-	a.CollectionIDs = collectionID
+	a.collectionIDs = collectionIDs
 }
 
+// ResetStatus is called to reset all the local variables for AttachmentMigrationManager.
 func (a *AttachmentMigrationManager) ResetStatus() {
 	a.lock.Lock()
 	defer a.lock.Unlock()
 
-	a.DocsProcessed.Set(0)
-	a.DocsChanged.Set(0)
-	a.CollectionIDs = nil
+	a.docsProcessed.Store(0)
+	a.docsChanged.Store(0)
+	a.docsFailed.Store(0)
+	a.collectionIDs = nil
+	a.migrationID = ""
```

Comment:
The changes to migrationID handling in this PR seem unrelated to the fix (not terminating on document deletion).  

Can you provide a bit of context on why it's needed?  Is it a bug related to DCP connection names when restarting a replication?

Reply:
I removed this code. I do not think it is right but it is outside the scope of this problem.